warehouse/query-core/src/main/java/datawave/query/jexl/lookups/ShardIndexQueryTableStaticMethods.java

```diff
@@ -587,44 +599,60 @@ public static RefactoredRangeDescription getRegexRange(Map.Entry<String,String>
         return getRegexRange(entry.getKey(), entry.getValue(), fullTableScanEnabled, metadataHelper, config);
     }
     
+    /**
+     * Determine whether a field =~ regex should be run against the reverse index or not.
+     * 
+     * @param analyzer
+     * @param fieldName
+     * @param metadataHelper
+     * @param config
+     * @return
+     * @throws TableNotFoundException
+     * @throws ExecutionException
+     */
     public static boolean shouldUseReverseIndex(JavaRegexAnalyzer analyzer, String fieldName, MetadataHelper metadataHelper, ShardQueryConfiguration config)
```

Comment:
there is similar logic in the IteratorBuildingVisitor, should these maybe be combined and extracted out?

Reply:
good call, I will take a look.

Reply:
After contemplating this one, I do not believe we can share code:  the IteratorBuildingVisitor which is for running against the field index does not have any reverse index to deal with whereas we do with the global indexes.

Reply:
After contemplating this one, I do not believe we can share code: the IteratorBuildingVisitor which is for running against the field index does not have any reverse index to deal with whereas we do with the global indexes.